### ISSUE TITLE
Moloni Plugin causes PHP Fatal Error (PHP8 - WHMCS 8.6.1)

### ISSUE THREAD

**Issue by maverick85** [moloni-pt/whmcs#1]
Daily cron fails to run due to `TypeError` related to the Moloni plugin.

Removing Moloni from the WHMCS installation allows cron to run correctly.

```
# !991
runuser -uapache -- /usr/bin/php -q /var/www/sites/proj1ty4/other/crons/cron.php -F -vvv

WHMCS Automation Task Utility: all
==================================

Queuing Tasks
-------------

 Force run any tasks: ignore "in progress" and "is due"
 Task queues ready

Executing Application Queue
---------------------------

  0/34 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0% < 1 sec/< 1 sec 34.0 MiB 
 Currency Exchange Rates
  1/34 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   2%  1 sec/34 secs 34.0 MiB 
 Product Pricing Updates
  2/34 [▓░░░░░░░░░░░░░░░░░░░░░░░░░░░]   5% 21 mins/6 hrs  36.0 MiB 
 Tenant Usage Metrics
  3/34 [▓▓░░░░░░░░░░░░░░░░░░░░░░░░░░]   8% 21 mins/4 hrs  36.0 MiB 
 Invoices
  4/34 [▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░░]  11% 21 mins/3 hrs  36.0 MiB 
 Late Fees
  5/34 [▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░]  14% 21 mins/2 hrs  36.0 MiB 
 Credit Card Charges
  6/34 [▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░]  17% 21 mins/2 hrs  42.0 MiB [WHMCS Application] ERROR: TypeError: Cannot access offset of type string on string in /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/Api/Customers.php:45
  Stack trace:
   #0 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/General.php(283): Moloni\Api\Customers::getByVat()
   #1 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/General.php(40): Moloni\General->verifyCustomer()
   #2 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/hooks.php(15): Moloni\General->createInvoice()
   #3 [internal function]: WHMCS\Utility\SafeInclude::{closure}()
   #4 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Hook/Manager.php(0): call_user_func()
   #5 /var/www/sites/proj1ty4/public_html/prod/vendor/illuminate/support/Facades/Facade.php(261): WHMCS\Hook\Manager->run()
   #6 /var/www/sites/proj1ty4/public_html/prod/includes/functions.php(0): Illuminate\Support\Facades\Facade::__callStatic()
   #7 /var/www/sites/proj1ty4/public_html/prod/includes/invoicefunctions.php(0): run_hook()
   #8 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Billing/Invoice.php(0): processPaidInvoice()
   #9 /var/www/sites/proj1ty4/public_html/prod/includes/ccfunctions.php(0): WHMCS\Billing\Invoice->addPayment()
   #10 /var/www/sites/proj1ty4/public_html/prod/includes/ccfunctions.php(0): captureCCPayment()
   #11 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Task/ProcessCreditCardPayments.php(0): ccProcessing()
   #12 [internal function]: WHMCS\Cron\Task\ProcessCreditCardPayments->__invoke()
   #13 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Scheduling/Task/AbstractTask.php(0): call_user_func_array()
   #14 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Scheduling/Task/AbstractTask.php(0): WHMCS\Scheduling\Task\AbstractTask->execute()
   #15 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Console/Command/AbstractCronCommand.php(0): WHMCS\Scheduling\Task\AbstractTask->run()
   #16 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Console/Command/AbstractCronCommand.php(0): WHMCS\Cron\Console\Command\AbstractCronCommand->executeCollection()
   #17 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Command/Command.php(298): WHMCS\Cron\Console\Command\AbstractCronCommand->execute()
   #18 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(1028): Symfony\Component\Console\Command\Command->run()
   #19 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
   #20 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
   #21 /var/www/sites/proj1ty4/offworld/crons/cron.php(0): Symfony\Component\Console\Application->run()
   #22 {main} {"exception":"[object] (TypeError(code: 0): Cannot access offset of type string on string at /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/Api/Customers.php:45)"} []
TypeError: Cannot access offset of type string on string in /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/Api/Customers.php:45

#0 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/General.php(283): Moloni\Api\Customers::getByVat()
#1 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/src/Moloni/General.php(40): Moloni\General->verifyCustomer()
#2 /var/www/sites/proj1ty4/public_html/prod/modules/addons/moloni/hooks.php(15): Moloni\General->createInvoice()
#3 [internal function]: WHMCS\Utility\SafeInclude::{closure}()
#4 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Hook/Manager.php(0): call_user_func()
#5 /var/www/sites/proj1ty4/public_html/prod/vendor/illuminate/support/Facades/Facade.php(261): WHMCS\Hook\Manager->run()
#6 /var/www/sites/proj1ty4/public_html/prod/includes/functions.php(0): Illuminate\Support\Facades\Facade::__callStatic()
#7 /var/www/sites/proj1ty4/public_html/prod/includes/invoicefunctions.php(0): run_hook()
#8 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Billing/Invoice.php(0): processPaidInvoice()
#9 /var/www/sites/proj1ty4/public_html/prod/includes/ccfunctions.php(0): WHMCS\Billing\Invoice->addPayment()
#10 /var/www/sites/proj1ty4/public_html/prod/includes/ccfunctions.php(0): captureCCPayment()
#11 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Task/ProcessCreditCardPayments.php(0): ccProcessing()
#12 [internal function]: WHMCS\Cron\Task\ProcessCreditCardPayments->__invoke()
#13 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Scheduling/Task/AbstractTask.php(0): call_user_func_array()
#14 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Scheduling/Task/AbstractTask.php(0): WHMCS\Scheduling\Task\AbstractTask->execute()
#15 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Console/Command/AbstractCronCommand.php(0): WHMCS\Scheduling\Task\AbstractTask->run()
#16 /var/www/sites/proj1ty4/public_html/prod/vendor/whmcs/whmcs-foundation/lib/Cron/Console/Command/AbstractCronCommand.php(0): WHMCS\Cron\Console\Command\AbstractCronCommand->executeCollection()
#17 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Command/Command.php(298): WHMCS\Cron\Console\Command\AbstractCronCommand->execute()
#18 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(1028): Symfony\Component\Console\Command\Command->run()
#19 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#20 /var/www/sites/proj1ty4/public_html/prod/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#21 /var/www/sites/proj1ty4/offworld/crons/cron.php(0): Symfony\Component\Console\Application->run()
#22 {main}
```

**Comment by maverick85** [comment]
The current fix for the issue is to remove the poorly coded addon:

```
[root@proj1ty4 public_html]# mv prod/modules/addons/moloni .
[root@proj1ty4 public_html]# runuser -uapache -- /usr/bin/php -q /var/www/sites/proj1ty4/other/crons/cron.php -F -vvv

WHMCS Automation Task Utility: all
==================================

Queuing Tasks
-------------

 Force run any tasks: ignore "in progress" and "is due"
 Task queues ready

Executing Application Queue
---------------------------

  0/34 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0% < 1 sec/< 1 sec 34.0 MiB 
 Currency Exchange Rates
  1/34 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   2% < 1 sec/< 1 sec 34.0 MiB 
 Product Pricing Updates
  2/34 [▓░░░░░░░░░░░░░░░░░░░░░░░░░░░]   5% 21 mins/6 hrs  36.0 MiB 
 Tenant Usage Metrics
  3/34 [▓▓░░░░░░░░░░░░░░░░░░░░░░░░░░]   8% 21 mins/4 hrs  36.0 MiB 
 Invoices
  4/34 [▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░░]  11% 21 mins/3 hrs  36.0 MiB 
 Late Fees
  5/34 [▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░]  14% 21 mins/2 hrs  36.0 MiB 
 Credit Card Charges
  6/34 [▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░░]  17% 21 mins/2 hrs  36.0 MiB 
 Invoice & Overdue Reminders
  7/34 [▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░]  20% 21 mins/1 hr   50.0 MiB 
 Domain Renewal Notices
  8/34 [▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░]  23% 21 mins/1 hr   50.0 MiB 
 Cancellation Requests
  9/34 [▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░]  26% 21 mins/1 hr   50.0 MiB 
 Overdue Suspensions
 10/34 [▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░]  29% 21 mins/1 hr   50.0 MiB 
 Overdue Terminations
 11/34 [▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░]  32% 21 mins/1 hr   50.0 MiB 
 Fixed Term Terminations
 12/34 [▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░]  35% 21 mins/1 hr   50.0 MiB 
 Inactive Tickets
 13/34 [▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░]  38% 21 mins/56 mins 50.0 MiB 
 Prune Ticket Attachments
 14/34 [▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░]  41% 21 mins/52 mins 50.0 MiB 
 Delayed Affiliate Commissions
 15/34 [▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░]  44% 21 mins/48 mins 50.0 MiB 
 Affiliate Reports
 16/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░]  47% 21 mins/45 mins 50.0 MiB 
 Process Email Campaigns
 17/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░]  50% 21 mins/43 mins 50.0 MiB 
 Process Email Queue
 18/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░]  52% 21 mins/40 mins 50.0 MiB 
 Email Marketer Rules
 19/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░]  55% 21 mins/38 mins 50.0 MiB 
 Credit Card Expiry Notices
 20/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░]  58% 21 mins/36 mins 50.0 MiB 
 SSL Sync
 21/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░]  61% 24 mins/39 mins 50.0 MiB 
 Server Usage Stats
 22/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░]  64% 24 mins/37 mins 50.0 MiB 
 Overage Billing Charges
 23/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░]  67% 24 mins/36 mins 50.0 MiB 
 Client Status Update
 24/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░]  70% 24 mins/34 mins 50.0 MiB 
 Domain Expiry
 25/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░]  73% 24 mins/33 mins 50.0 MiB 
 Ticket Escalation Rules
 26/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░]  76% 24 mins/32 mins 50.0 MiB 
 Data Retention Pruning
 27/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░]  79% 24 mins/30 mins 50.0 MiB 
 SSL Certificate Reissues
 28/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░]  82% 24 mins/29 mins 50.0 MiB 
 Update Server Usage
 29/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░]  85% 24 mins/28 mins 50.0 MiB 
 Update Server Meta Data
 30/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░]  88% 24 mins/27 mins 50.0 MiB 
 WHMCS Updates
 31/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░]  91% 24 mins/27 mins 52.0 MiB 
 Run Jobs Queue
 32/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░]  94% 24 mins/26 mins 52.0 MiB 
 Domain Transfer Status Synchronisation
 33/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░]  97% 24 mins/25 mins 52.0 MiB 
 Domain Status Synchronisation
 34/34 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% 27 mins/27 mins 60.0 MiB


 Sending Daily Cron Digest email

Executing System Queue
----------------------

 4/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% < 1 sec/< 1 sec 62.0 MiB

                                                                                                                        
 [OK] Completed                                                                                                         
                                                                                                                        
```

**Comment by moloni-pt** [comment]
Dear customer, 

Thank you for reporting this error. 
Currently this issue has been fixed on version 3.3.5 where you can find at https://github.com/moloni-pt/whmcs/releases/tag/v3.3.5
